envoy/tracing/trace_context.h

```diff
@@ -14,15 +15,51 @@ namespace Tracing {
  * Protocol-independent abstraction for traceable stream. It hides the differences between different
  * protocol and provides tracer driver with common methods for obtaining and setting the tracing
  * context.
- *
- * TODO(wbpcode): A new interface should be added to obtain general traceable stream information,
- * such as host, RPC method, protocol identification, etc. At the same time, a new interface needs
- * to be added to support traversal of all trace contexts.
  */
 class TraceContext {
 public:
   virtual ~TraceContext() = default;
 
+  using IterateCallback = std::function<bool(absl::string_view key, absl::string_view val)>;
+
+  /**
+   * Get context protocol.
+   *
+   * @return A string view representing the protocol of the traceable stream behind the context.
+   */
+  virtual absl::string_view contextProtocol() const PURE;
```

Comment:
nit: remove `context` from all of these method names since it's redundant with the class name.

Reply:
I also hope to do this, but this may cause conflicts between the method names and the method names of the HeaderMap itself.

Reply:
Perhaps we should not let RequestHeaderMap directly inherit TraceContext, but should create a new RequestHeaderMap Wrapper type to inherit TraceContext. 🤔

Reply:
OK. However, after modification, the names of these methods may be confused with some methods in RequestHeaderMap. For example, the `protocol` from TraceContext and the `Protocol` method of RequestHeaderMap itself. We need to add more comments to avoid incorrect use.